packages/mostly-equal/src/mostly-equal.ts

```diff
@@ -0,0 +1,280 @@
+    if (isExpectValuesInfo(item)) {
+      if (valueErrors.has(item.uniqueSymb) && valueErrors.get(item.uniqueSymb)?.has(item)) {
+        return [safePrint(item.value, 0), valueErrors.get(item.uniqueSymb)!.get(item)!];
+      } else {
+        return [safePrint(item.value, 0)];
+      }
+    }
+    return item;
+  });
+};
+
+const tryExpectVal = (
+  expected: ExpectValue<any>,
+  actual: any,
+  depth: number,
+  path: LookupPath,
+  passedMap: Map<any, LookupPath>,
+  passedSet: Set<any>,
+  existsInParent: boolean
+): ErrorOrTextOrExpect => {
+  let matcherRes: undefined | string | void = undefined;
+  try {
+    matcherRes = expected.expectMethod(actual, existsInParent, path);
+  } catch (err) {
+    return [safePrint(actual, depth, passedMap, passedSet, path), anyToError(err)];
+  }
+  if (matcherRes !== undefined && matcherRes !== null) {
+    return [matcherRes.toString()];
+  }
+  return [safePrint(actual, depth, passedMap, passedSet, path)];
+};
+
+export const errorString: (
+  expected: any,
+  actual: any,
+  depth: number,
+  path: LookupPath,
+  passedMap: Map<any, LookupPath>,
+  passedSet: Set<any>
+) => ErrorOrTextOrExpect = (expected, actual, depth, path, passedMap, passedSet) => {
+  if (isExpectVal(expected)) {
+    return tryExpectVal(expected, actual, depth, path, passedMap, passedSet, true);
+  }
+
+  if (isExpectValues(expected)) {
+    return [
+      {
+        uniqueSymb: expected,
+        value: actual,
+        fieldDefinedInParent: true,
+        path,
+      },
+    ];
+  }
+
+  if (expected === actual) {
+    return [safePrint(actual, depth, passedMap, passedSet, path)];
+  }
+  if (Array.isArray(expected)) {
+    if (Array.isArray(actual)) {
+      if (actual.length !== expected.length) {
+        return [
+          anyToError(`expected length ${expected.length} but got ${actual.length}`),
+          safePrint(actual, depth, passedMap, passedSet, path),
+        ];
+      }
+
+      const res: ErrorOrTextOrExpect = ['[ \n', spaces(depth)];
+      const childSet = registerChildSet(actual, path, passedMap, passedSet);
+      for (let i = 0; i < actual.length; i++) {
+        res.push(...errorString(expected[i], actual[i], depth + 1, [...path, i], passedMap, childSet), ',');
+      }
+      res.push(`\n${spaces(depth)}]\n${spaces(depth)}`);
+      return res;
+    } else {
+      return [
+        anyToError(
+          `expected ${safePrint(expected, 0, passedMap, passedSet, path)} but got ${safePrint(
+            actual,
+            0,
+            passedMap,
+            passedSet,
+            path
+          )}`
+        ),
+        safePrint(actual, depth),
+      ];
+    }
+  }
+
+  if (expected instanceof Object) {
+    if (actual instanceof Object) {
+      const res: ErrorOrTextOrExpect = [`\n${spaces(depth)}{`];
+      const childSet = registerChildSet(actual, path, passedMap, passedSet);
+
+      const allNames = [...new Set([...Object.keys(expected), ...Object.keys(actual)])];
```

Comment:
leave it as Set. the code uses for-const-of on it, so no need for the array destructuring

Reply:
done